### DISCUSSION TITLE
One-Click (unattended) install

### DISCUSSION THREAD

**Discussion by sraedler** [YerongAI/Office-Tool#523] [Q&A] [answered]
I'm trying to create a unattended offline install for Office 2021 or 2019. Both I cannot figure out.

What I did:

1. Download the image using the attached configuration
2. I try to call the console helper with the following command:

`"Office Tool Plus.Console.exe" deploy /addProduct ProPlus2021Volume_de-de_Volume_Outlook,Word,Excel,Powerpoint /channel PerpetualVL2021 /sourcePath C:\Users\User\Desktop\OfficeTool\Office /version 16.0.14332.20324`

The sourcePath is different because I'm testing on another machine.

Edit: Both Machines are running Windows 11

**Comment by YerongAI** [comment, marked as answer]
There is some problems with your command:

1. Your addProduct command is invalid, the correct one is: `/addProduct ProPlus2021Volume_de-de_Outlook,Word,Excel,PowerPoint`.
2. There may be something wrong with the `/sourcePath `, if your source path is **C:\Users\User\Desktop\OfficeTool\Office\Data\16.0.xxxxx.xxxxx**, the right parameter is `C:\Users\User\Desktop\OfficeTool`. And if the folder name contains spaces, you should write them down like: `"C:\Users\User\Desktop\Office Tool"`.
3. Also it's better to set the Office edition by `/clientEdition`, or 32-bit of Office will be installed.

More details of deploy commands can be found at [https://help.coolhub.top/others/#deploy-commands](https://help.coolhub.top/others/#deploy-commands)

If any questions please let us know.

**Reply by sraedler** [reply]
Thank you very much for your suggestion. I have updated the command and also tried to change the directory in case there are problems with permissions.
However, I still have problems. In the attachment you can see my executed command.
Can you please give me the correct command?


![image](https://user-images.githubusercontent.com/11181512/174472750-477ba6ed-3c6b-409b-ac75-89ca3e9df930.png)

**Reply by YerongAI** [reply]
Oh, you need to set the Office edition by `/edition 64`, or 32-bit of Office will be installed by default. Sorry for my mistake, I will update the document and point them out. Thanks for your feedback.

**Reply by sraedler** [reply]
Also with the following it is not working:

`"Office Tool Plus.Console.exe" deploy /addProduct ProPlus2021Volume_de-de_Outlook,Word,Excel,Powerpoint /channel PerpetualVL2021 /sourcePath C:\OfficeTool /version 16.0.14332.20324 /edition 64`

**Reply by YerongAI** [reply]
What is the error window showed? And if possible, could you please collect installer logs and send to me? You can collect logs via Office Tool Plus (on toolbox page), it's not required installer run again.

**Reply by sraedler** [reply]
The Error Code: 30182-2016 (2)

In the following Zip, the Log files. 

[Office_Log.zip](https://github.com/YerongAI/Office-Tool/files/8935996/Office_Log.zip)


Thanks for your support!

**Reply by YerongAI** [reply]
Omg, very sorry I have give you the wrong command again 😢😭, the right one should be `"Office Tool Plus.Console.exe" deploy /addProduct ProPlus2021Volume_de-de_Outlook,Word,Excel,PowerPoint /channel PerpetualVL2021 /sourcePath C:\OfficeTool /version 16.0.14332.20324 /clientEdition 64`.

More details of deploy commands can be found at [https://help.coolhub.top/others/#deploy-commands](https://help.coolhub.top/others/#deploy-commands)

**Reply by sraedler** [reply]
Thank you very much for your support!

It works with the suggested command line. 

Could you please add a column to the documentation indicating which parameters are mandatory and which are optional?

**Reply by YerongAI** [reply]
> Thank you very much for your support!
> 
> It works with the suggested command line. 
> 
> Could you please add a column to the documentation indicating which parameters are mandatory and which are optional?
> 

Sure, I will do it as fast as possible.
